staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -889,8 +896,15 @@ func (r *Request) requestPreflightCheck() error {
 }
 
 func (r *Request) newHTTPRequest(ctx context.Context) (*http.Request, error) {
+	var body io.Reader
+	if r.body != nil {
+		body = r.body
+	} else if r.bodyBytes != nil {
+		body = bytes.NewReader(r.bodyBytes)
+	}
```

Comment:
this is the key, reading from the slice is thread safe and we create a new body each time, meanwhile the opaque reader is plumbed down